store/sqlstore/upgrade.go

```diff
@@ -1000,6 +1000,30 @@ func upgradeDatabaseToVersion533(sqlStore *SqlStore) {
 
 func upgradeDatabaseToVersion534(sqlStore *SqlStore) {
 	// if shouldPerformUpgrade(sqlStore, Version5330, Version5340) {
+	sqlStore.CreateColumnIfNotExistsNoDefault("ChannelMembers", "MentionCountRoot", "bigint", "bigint")
+	updateMentionCountRootQuery := `
+	UPDATE ChannelMembers INNER JOIN (
+		SELECT Threads.ChannelId,COALESCE(threadmemberships.UserId,'') AS UserId, COALESCE(SUM(UnreadMentions),0) AS UnreadMentions FROM Channels
+		LEFT JOIN Threads ON Threads.ChannelId = Channels.Id
+		LEFT Join ThreadMemberships ON Threads.PostId = ThreadMemberships.PostId
+		Group BY Threads.ChannelId,ThreadMemberships.UserId
+	) AS q ON q.ChannelId = ChannelMembers.ChannelId AND (q.UserId=ChannelMembers.UserId OR q.UserId='')
+	SET MentionCountRoot = GREATEST(ChannelMembers.MentionCount - q.UnreadMentions, 0)`
```

Comment:
I feel that this query is just too heavy to be part of a synchronous migration process. I am yet to see this return on our biggest MySQL dataset. I imagine it could take hours to execute given the amount of data involved in the subquery.

Reply:
Purely for curiosity this has finally returned:

```
mysql> UPDATE ChannelMembers
    -> INNER JOIN
    ->   (SELECT Threads.ChannelId,
    ->           COALESCE(ThreadMemberships.UserId, '') AS UserId,
    ->           COALESCE(SUM(UnreadMentions), 0) AS UnreadMentions
    ->    FROM Channels
    ->    LEFT JOIN Threads ON Threads.ChannelId = Channels.Id
    ->    LEFT JOIN ThreadMemberships ON Threads.PostId = ThreadMemberships.PostId
    ->    GROUP BY Threads.ChannelId,
    ->             ThreadMemberships.UserId) AS q ON q.ChannelId = ChannelMembers.ChannelId
    -> AND (q.UserId=ChannelMembers.UserId
    ->      OR q.UserId='')
    -> SET MentionCountRoot = GREATEST(ChannelMembers.MentionCount - q.UnreadMentions, 0);
Query OK, 3257748 rows affected (3 hours 2 min 16.88 sec)
Rows matched: 3257748  Changed: 3257748  Warnings: 0
```